### ISSUE TITLE
Airsonic start up failing after update with UnsatisfiedDependencyException

### ISSUE THREAD

**Issue by khers** [airsonic-advanced/airsonic-advanced#782]
## Problem description

Airsonic is failing to start after updating to latest war

```
2022-02-01 18:06:54.483  INFO --- l.l.StandardLockService                  : Successfully released change log lock
2022-02-01 18:06:54.485  WARN --- ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: a FOREIGN KEY constraint already exists on the set of columns: URA_U_FK in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ura_u_fk FOREIGN KEY (username) REFERENCES PUBLIC.users (username) ON DELETE CASCADE] [Failed SQL: (-5528) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ura_u_fk FOREIGN KEY (username) REFERENCES PUBLIC.users (username) ON DELETE CASCADE]
2022-02-01 18:06:58.748 ERROR --- o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
     Reason: liquibase.exception.DatabaseException: a FOREIGN KEY constraint already exists on the set of columns: URA_U_FK in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ura_u_fk FOREIGN KEY (username) REFERENCES PUBLIC.users (username) ON DELETE CASCADE] [Failed SQL: (-5528) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ura_u_fk FOREIGN KEY (username) REFERENCES PUBLIC.users (username) ON DELETE CASCADE]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:768) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.inject(AutowiredAnnotationBeanPostProcessor.java:720) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1413) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:213) ~[spring-beans-5.3.8.jar!/:5.3.8]
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1160) ~[spring-context-5.3.8.jar!/:5.3.8]
        at org.springframework.ui.context.support.UiApplicationContextUtils.initThemeSource(UiApplicationContextUtils.java:58) ~[spring-context-5.3.8.jar!/:5.3.8]
...
```
I can get the rest of the stack trace if necessary

### Steps to reproduce

1. Start with a working install of Edge Release 11.0.0-SNAPSHOT.20220110133315
2. Update war to Edge Release 11.0.0-SNAPSHOT.20220201172921
3. Startup will crash

## System information

 * **Airsonic version**: Edge Release 11.0.0-SNAPSHOT.20220201172921
 * **Operating system**: Ubuntu 20.04
 * **Java version**: openjdk version "13.0.7" 2021-04-20
 * **Proxy server**: Apache from Ubuntu 20.04
 * **Client**: Firefox 96
 * **Language**: en_US

**Comment by khers** [comment]
Edge Release 11.0.0-SNAPSHOT.20220129204202 Is the first release that fails to start. Unfortunately, I didn't see the "non-backwards compatible before jumping over this one and am in a bit of a complicated state. I can roll back and get the service up, but podcasts no longer work.

**Comment by Gronkdalonka** [comment]
looks like  #763

**Comment by monik3r** [comment]
I'm having a similar issue, no stack trace, but the webserver is failing to start. 
```022-02-01 20:41:42.475  INFO --- org.airsonic.player.Application          : Starting Application using Java 14.0.2 on a3a1482edace with PID 1 (/app/WEB-INF/classes started by au in /var)
2022-02-01 20:41:42.477  INFO --- org.airsonic.player.Application          : No active profile set, falling back to default profiles: default
2022-02-01 20:41:49.308  INFO --- l.database.core.HsqlDatabase             : Set default schema name to PUBLIC
2022-02-01 20:41:49.417  INFO --- l.l.StandardLockService                  : Successfully acquired change log lock
2022-02-01 20:41:50.231  INFO --- l.c.StandardChangeLogHistoryService      : Reading from PUBLIC.DATABASECHANGELOG
2022-02-01 20:41:50.395  INFO --- l.l.StandardLockService                  : Successfully released change log lock
2022-02-01 20:41:50.405  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:42:00.411  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:42:10.421  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:42:20.430  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:42:30.439  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:42:40.448  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:42:50.458  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:43:00.466  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
2022-02-01 20:43:10.473  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....```

**Comment by khers** [comment]
> I'm having a similar issue, no stack trace, but the webserver is failing to start.
> 
> ```
> 2022-02-01 20:41:42.477  INFO --- org.airsonic.player.Application          : No active profile set, falling back to default profiles: default
> 2022-02-01 20:41:49.308  INFO --- l.database.core.HsqlDatabase             : Set default schema name to PUBLIC
> 2022-02-01 20:41:49.417  INFO --- l.l.StandardLockService                  : Successfully acquired change log lock
> 2022-02-01 20:41:50.231  INFO --- l.c.StandardChangeLogHistoryService      : Reading from PUBLIC.DATABASECHANGELOG
> 2022-02-01 20:41:50.395  INFO --- l.l.StandardLockService                  : Successfully released change log lock
> 2022-02-01 20:41:50.405  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:42:00.411  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:42:10.421  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:42:20.430  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:42:30.439  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:42:40.448  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:42:50.458  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:43:00.466  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....
> 2022-02-01 20:43:10.473  INFO --- l.l.StandardLockService                  : Waiting for changelog lock....```
> ```

I don't think this is the same issue, you might want to open a new one.

**Comment by randomnicode** [comment]
@monik3r your db is waiting on another thread that has the db open (maybe another airsonic instance) or has the lock row in the db set. either close the other instance, or delete the row and retrry

**Comment by randomnicode** [comment]
@khers this is a non backward compatible change, rolling back will run into these issues as the old code is querying columns like `path` that don't exist anymore.

the actual issue is the failed migration:
```
Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::add-user-rating-pk::anon:
Reason: liquibase.exception.DatabaseException: a FOREIGN KEY constraint already exists on the set of columns: URA_U_FK in statement [ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ura_u_fk FOREIGN KEY (username) REFERENCES PUBLIC.users (username) ON DELETE CASCADE] [Failed SQL: (-5528) ALTER TABLE PUBLIC.user_rating ADD CONSTRAINT ura_u_fk FOREIGN KEY (username) REFERENCES PUBLIC.users (username) ON DELETE CASCADE]
```

do you still have this issue with the latest?

**Comment by randomnicode** [comment]
can you post your db also?

we're looking specifically for the databasechangelock and user_rating table (which will pull in users and media_file)

**Comment by monik3r** [comment]
@khers thanks, I'll move that to another issue

**Comment by khers** [comment]
The latest commit fixed me as well.

**Comment by khers** [comment]
@randomnicode Thanks for all the hard work on this project, I hope the bug reports are helpful and not an annoyance.
